packages/datadog-plugin-winston/test/index.spec.js

```diff
@@ -218,6 +218,27 @@ describe('Plugin', () => {
               expect(await logServer.logPromise).to.include(dd)
             })
           }
+
+          it('should overwrite any existing "dd" property', async () => {
```

Comment:
It should probably _not_ overwrite as it could break an existing log.

Reply:
The behaviour prior to #1801 was that we _would_ overwrite any pre-existing `dd` properties, and then completely delete them. If we cannot override the property (even virtually using a proxy), then we cannot do log injection at all, unless we find some other "nearby" unused property name (e.g. the first unused property in `dd`, `dd1`, `dd2`, etc.) and ensure all the log correlation tools work with these "nearby" property names.

That is to say, if the previous behaviour was a bug, then it needs to be addressed at a whole-system level, and not just in this library.

Reply:
should this come into the discussion of moving things into WeakMaps or Symbols as a separate PR then?

Reply:
> The behaviour prior to #1801 was that we would overwrite any pre-existing dd properties, and then completely delete them.

This was a hacky workaround and shouldn't be used as the basis of what should be done with the proxy. This is especially true if a test didn't exist, meaning the behaviour was undefined.

> should this come into the discussion of moving things into WeakMaps or Symbols as a separate PR then?

For this use case it wouldn't work because the idea is explicitly to inject into the log entry of the user.

Reply:
Our options are:

1. A similar hacky workaround to what we have today, but with the bonus of not messing with the original object (i.e. the proxy solution, or copy properties to new objects).
2. Change the entire system to have fallback "nearby" property names.
3. Fallback to skipping log injection if the `dd` property exists.

I'm not sure there are other options. Option 1 seemed to be the only one that

Reply:
I guess it will be easier in any case to preserve the current behaviour, so option 1 is fine then.